garak/generators/bedrock.py

```diff
@@ -0,0 +1,251 @@
+            if env_region:
+                logging.info(f"Using AWS region from environment: {env_region}")
+                self.region = env_region
+        
+        return super()._validate_env_var()
+
+    def _load_client(self):
+        """Load and configure the boto3 bedrock-runtime client.
+
+        Uses boto3's standard credential chain for authentication.
+        """
+        try:
+            import boto3
+        except ImportError:
+            raise ImportError(
+                "boto3 is required for the Bedrock generator. "
+                "Install it with: pip install boto3"
+            )
+
+        self.client = boto3.client(
+            service_name="bedrock-runtime",
+            region_name=self.region,
+        )
+
+        logging.info(f"Loaded boto3 bedrock-runtime client for region {self.region}")
+
+    def _clear_client(self):
+        """Clear the boto3 client to enable object pickling."""
+        if hasattr(self, "client"):
+            self.client = None
+
+    def __getstate__(self):
+        """Prepare object for pickling by clearing the boto3 client."""
+        self._clear_client()
+        return self.__dict__
+
+    def __setstate__(self, state):
+        """Restore object from pickle and reload the boto3 client."""
+        self.__dict__.update(state)
+        self._load_client()
+
+    @backoff.on_exception(
+        backoff.fibo,
+        garak.exception.GarakBackoffTrigger,
+        max_value=70,
+    )
+    def _call_model(
+        self, prompt: Conversation, generations_this_call: int = 1
+    ) -> List[Union[Message, None]]:
+        """Call the Bedrock model using the Converse API.
+
+        Args:
+            prompt: Conversation object containing the prompt turns
+            generations_this_call: Number of generations to request (currently only 1 is supported)
+
+        Returns:
+            List of Message objects containing the generated text, or [None] on error
+        """
+        if self.client is None:
+            self._load_client()
+
+        messages = []
+        for turn in prompt.turns:
+            if turn.role in ["user", "assistant"]:
+                messages.append(
+                    {"role": turn.role, "content": [{"text": turn.content.text}]}
+                )
+
+        if not messages:
+            logging.error("No valid messages to send to Bedrock")
+            return [None]
+
+        inference_config = {}
+        if self.temperature is not None:
+            inference_config["temperature"] = float(self.temperature)
+        if hasattr(self, "max_tokens") and self.max_tokens is not None:
+            inference_config["maxTokens"] = int(self.max_tokens)
+        if self.top_p is not None:
+            inference_config["topP"] = float(self.top_p)
+        if self.stop and isinstance(self.stop, list) and len(self.stop) > 0:
+            inference_config["stopSequences"] = self.stop
```

Comment:
The type expectations for `stopSequences` may be appropriate to have validated in `__init__` as an early notification to the user that the configuration passed is invalid. Simply suppressing a improperly formatted configuration might lead users to rely on results that did not take into account the intended option.

That might allow this check to be made more efficient as:

```suggestion
        if self.stop:
            inference_config["stopSequences"] = self.stop
```

Reply:
As a side note, this is a reason the project should consider expanding `DEFAULT_PARAMS` definitions with type hints and descriptions to offload this type of validation to common tooling in `Configurable`.